Makefile

```diff
@@ -112,59 +115,94 @@ demo: ## Execute end-to-end demo
 
 # spin-up local environment
 .PHONY: deploy-full-local-setup
-deploy-full-local-setup: ## Deploy full local multicluster setup (k3d >= 4.2.0)
-	$(call create-local-cluster,$(CLUSTER_GSLB1))
-	$(call create-local-cluster,$(CLUSTER_GSLB2))
+deploy-full-local-setup: ensure-cluster-size ## Deploy full local multicluster setup (k3d >= 4.2.0)
+	@echo "\n$(YELLOW)Creating $(CLUSTERS_NUMBER) k8s clusters$(NC)"
+	@for c in $(CLUSTER_IDS); do \
+		$(MAKE) create-local-cluster CLUSTER_NAME=$(CLUSTER_NAME)$$c ;\
+	done
 
-	$(call deploy-local-cluster,$(CLUSTER_GSLB1),$(CLUSTER_GSLB2),$(VERSION),,'k8gb/k8gb')
-	$(call deploy-local-cluster,$(CLUSTER_GSLB2),$(CLUSTER_GSLB1),$(VERSION),$(CLUSTER_GSLB2_HELM_ARGS),'k8gb/k8gb')
-
-	@echo "\n$(YELLOW)Deploy test apps $(NC)"
-	$(call deploy-test-apps,$(CLUSTER_GSLB1))
-	$(call deploy-test-apps,$(CLUSTER_GSLB2))
+	$(MAKE) deploy-stable-version
```

Comment:
Same question here, why use external make call instead of function?

Reply:
`deploy-stable-version` was actually a "standalone target" even before (so it can't be a function), it's being called from `upgrade-testing.yaml`. What I did here was little bit of code de-duplication. The thing is that some targets needs to be self-contained from start to end (`deploy-full-local-setup` - one liner that should do everything including creation of k3d clusters) but some (`deploy-stable-version`) assume the cluster to be created (in our testing the cluster is created by k3d action)

it's a negative side-effect of mixing tasks that might be convenient for a user with targets that are required by CI. But if those two concerns live in the same file, this is a reusable (/"DRY") thing to do.